Extensions/BitmapText/bitmaptextruntimeobject.ts

```diff
@@ -0,0 +1,296 @@
+namespace gdjs {
+  /** Base parameters for {@link gdjs.BitmapTextRuntimeObject} */
+  export type BitmapTextObjectDataType = {
+    /** The base parameters of the Bitmap Text */
+    content: {
+      /** The opacity of the Bitmap Text */
+      opacity: float;
+      /** Content of the text */
+      text: string;
+      /** The tint of the text */
+      tint: Array<number>;
+      /** The bitmap font file for the text */
+      bitmapFontFile: string;
+      /** The atlas image file for the text */
+      bitmapAtlasFile: string;
+      /** The scale of the text */
+      scale: float;
+      /** Activate word wrap if set to true */
+      wordWrap: boolean;
+      /** Wrapping with from custom size properties */
+      wrappingWidth: float;
+      /** Alignment of the text: "left", "center" or "right" */
+      align: 'left' | 'center' | 'right';
+    };
+  };
+  export type BitmapTextObjectData = ObjectData & BitmapTextObjectDataType;
+
+  /**
+   * Displays a bitmap text without losing in quality on edges
+   * The object use files generated in a external editor like bmFont.
+   */
+  export class BitmapTextRuntimeObject extends gdjs.RuntimeObject {
+    _opacity: float;
+    _text: string;
+    /** color in format [r, g, b], where each component is in the range [0, 255] */
+    _tint: integer[];
+    _fontFamily: string;
+    _bitmapFontFile: string;
+    _bitmapAtlasFile: string;
+    _scale: number;
+    _wordWrap: boolean;
+    _wrappingWidth: float;
+    _align: string;
+
+    _renderer: gdjs.BitmapTextRuntimeObjectRenderer;
+
+    /**
+     * @param runtimeScene The scene the object belongs to.
+     * @param objectData The object data used to initialize the object
+     */
+    constructor(runtimeScene: gdjs.RuntimeScene, objectData: BBTextObjectData) {
+      super(runtimeScene, objectData);
+
+      this._opacity = objectData.content.opacity;
+      this._text = objectData.content.text;
+      this._tint = gdjs.hexToRGBColor(objectData.content.tint);
+      this._bitmapFontFile = objectData.content.bitmapFontFile; // fnt/xml files
+      this._bitmapAtlasFile = objectData.content.bitmapAtlasFile; // texture file used with fnt/xml (bitmap font file)
+      this._scale = objectData.content.scale;
+      this._wordWrap = objectData.content.wordWrap;
+      this._wrappingWidth = 0;
+      this._align = objectData.content.align;
+
+      // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
```

Comment:
Am I missing something, or is there a missing `this._renderer = ...`? 🤔

Reply:
I can't said until the typescript conversion isn't complete.
Test was done on javascript, [598f549dee24a40d4885ba3858b9d164475b1b5a](https://github.com/4ian/GDevelop/pull/1894/commits/598f549dee24a40d4885ba3858b9d164475b1b5a) is the latest commit in js and it was working.
Scripts for convert js to ts wasn't working well for me without loose some lines, unfortunatly manually it not better than scripts if i loose also some lines.

Reply:
Maybe it worth for you to use the latest commit working in JS and convert it with your tools.

Reply:
Ok I'll check that :) Remember that JS is valid TypeScript, so you could go progressively by just renaming your .js file to .ts and start "upgrading" the code from here.
But as you said I'll start from the last JS commit and convert it with my scripts :)

Reply:
Actually I can't easily start back from the JS commit, so I'll have to use the TS version. In the future prefer to always keep something working - even if it's not complete or need manually work, as it will save me some time having to find what's broken.